safe/report/test/test_expressions.py

```diff
@@ -0,0 +1,262 @@
+    minimum_needs_section_notes_element)
+from safe.report.expressions.map_report import (
+    legend_title_header_element,
+    disclaimer_title_header_element,
+    disclaimer_text_element,
+    information_title_header_element,
+    time_title_header_element,
+    caution_title_header_element,
+    caution_text_element,
+    source_title_header_element,
+    analysis_title_header_element,
+    version_title_header_element,
+    reference_title_header_element,
+    unknown_source_text_element,
+    aggregation_not_used_text_element,
+    inasafe_logo_black_path,
+    inasafe_logo_white_path,
+    north_arrow_path,
+    organisation_logo_path)
+
+__copyright__ = "Copyright 2017, The InaSAFE Project"
+__license__ = "GPL version 3"
+__email__ = "info@inasafe.org"
+__revision__ = '$Format:%H$'
+
+default_params = []
+feature = None
+parent = None
+
+
+class TestExpressions(unittest.TestCase):
+    """Test Expressions.
+
+    .. versionadded:: 4.3
+    """
+
+    def get_function(self, function_name):
+        """Get an expression function which is already loaded in qgis.
+
+        :param function_name: Name of the function.
+        :type function_name: str
+
+        :return: The function.
+        :rtype: QgsExpressionFunction
+
+        .. versionadded:: 4.3
+        """
+        for func in QgsExpression.Functions():
+            if func.name() == function_name:
+                return func
+        return None
+
+    def evaluate(self, func, expected_result, params=None):
+        """Evaluate the result of a function.
+
+        :param func: The function.
+        :type func: QgsExpressionFunction
+
+        :param expected_result: The expected result.
+        :type expected_result: str
+
+        :param params: The parameters that used by the function.
+        :type params: list
+
+        .. versionadded:: 4.3
+        """
+        if not None:
+            params = default_params
+
+        actual_result = func.funcV2(params, feature, parent)
+        self.assertEqual(actual_result, expected_result)
+
+    def test_map_report_expressions(self):
+        """Test all expressions in the map report expressions.
+
+        .. versionadded:: 4.3
+        """
+
+        # legend_title_header_element
+        expected_result = u'Legend'
+        self.evaluate(legend_title_header_element, expected_result)
+
+        # disclaimer_title_header_element
+        expected_result = u'Disclaimer'
+        self.evaluate(disclaimer_title_header_element, expected_result)
+
+        # disclaimer_text_element
+        expected_result = (
+            u'InaSAFE has been jointly developed by the Indonesian '
+            u'Government-BNPB, the Australian Government, '
+            u'the World Bank-GFDRR and independent contributors. '
+            u'These agencies and the individual software developers of '
+            u'InaSAFE take no responsibility for the correctness of outputs '
+            u'from InaSAFE or decisions derived as a consequence.')
```

Comment:
Can we take these strings from definitions?
If some not python developers come to update definitions, it doesn't break tests which are only strings.
Same for other tests

Reply:
so we don't want them to update the test as they update the definition?

Reply:
Well, they are not aware of these tests until you launch them on travis. (even me)

I think it's a little bit annoying when you change only one character in a string, it doesn't make sense to check the content of the string, especially these strings. You should be able to fix a typo without breaking tests. The test should only check the logic, that it returns a string with some content, not checking the actual content of the string.

To sum up, updating strings like this shouldn't break test.

My opinion about what is a good test. My 2 cents ;-)

Reply:
+1 for @Gustry 's comment